enterprise_learner_portal/api/v1/views.py

```diff
@@ -100,7 +100,8 @@ def patch(self, request):
             course_id=course_id
         )
 
-        enterprise_enrollment.marked_done = marked_done
+        # TODO: For now, this makes the change backward compatible, we will change this to true boolean support
+        enterprise_enrollment.marked_done = marked_done.lower() == 'true'
```

Comment:
For context the problem is this was never resolving to be a boolean, or was it just never a boolean in the case of learner-portal's requests to this endpoint.

For backwards compatibility, we'd probably want to do some typechecking here instead after we do the `.get()` for `marked_done`. If we get a boolean back, great, let's keep it, if we get a string back, do the `.lower() == 'true'` check.

Reply:
I am pretty sure we always got a string here, previously we got a 'True' or 'False' and now we get 'true' or 'false' because of a change inthe frontend to send boolean fields which get turned into strings 'true' or 'false' (since a text/plain is used here when sedning from client) ... but the field being assigned to was a BooleanField so I think the first case just worked while the latter needed explicit handling ... I am ~95% on that

Reply:
also, soon this will change to be all boolean on both client and server side if I do this right ...

Reply:
and just to be sure I tested by interfering with the payload by changing the options['marked_done'] to 'True' or 'False' instead of 'true' or 'false' and both seem to work correctly...